addons/html_builder/static/src/website_sale/product_image_option.xml

```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+
+
+<t t-inherit="html_builder.ReplaceMediaOption" t-inherit-mode="extension">
+    <xpath expr="//BuilderRow[BuilderButton[@action=&quot;'replaceMedia'&quot;]]" position="after">
+        <BuilderRow label.translate="Re-order">
+            <BuilderButtonGroup preview="false" action="'setPosition'">
+                <BuilderButton icon="'fa-angle-double-left'" title.translate="Move to first"  actionValue="'first'"/>
+                <BuilderButton icon="'fa-angle-left'" title.translate="Move to previous" actionValue="'left'"/>
+                <BuilderButton icon="'fa-angle-right'" title.translate="Move to next" actionValue="'right'"/>
+                <BuilderButton icon="'fa-angle-double-right'" title.translate="Move to last" actionValue="'last'"/>
+            </BuilderButtonGroup>
+        </BuilderRow>
+    </xpath>
+
+    <xpath expr="//BuilderRow[BuilderButton[@action=&quot;'replaceMedia'&quot;]]" position="inside">
+        <BuilderButton
+        action="'removeMedia'"
+        preview="false"
+        type="'danger'"
+        className="'flex-grow-1'"        
+        label.translate="Remove" />
+    </xpath>
+</t>
+
+</templates>
```

Comment:
General note: you probably need something equivalent to `isProductPageImage` (see `_computeWidgetVisibility`) otherwise, the "Re-order" and "Remove" options will appear on other field as well such (e.g. the logo).